rmf_building_sim_gz_plugins/src/lift.cpp

```diff
@@ -122,185 +98,547 @@ class IGNITION_GAZEBO_VISIBLE LiftPlugin
     return payloads;
   }
 
-public:
-  LiftPlugin()
+  void initialize_components(EntityComponentManager& ecm)
   {
-    // TODO init ros node
-    // Do nothing
+    ecm.Each<components::Lift>([&](const Entity& entity,
+      const components::Lift* lift_comp) -> bool
+      {
+        const auto& lift = lift_comp->Data();
+        auto cabin_joint_entity =
+        Model(entity).JointByName(ecm, lift.cabin_joint);
+        enableComponent<components::AxisAlignedBox>(ecm, entity);
+        enableComponent<components::JointPosition>(ecm, cabin_joint_entity);
+        ecm.CreateComponent<components::JointVelocityCmd>(cabin_joint_entity,
+        components::JointVelocityCmd({0.0}));
+
+        LiftCommand lift_command;
+        lift_command.request_type = LiftRequest::REQUEST_AGV_MODE;
+        // Set the initial floor
+        const auto target_it = lift.floors.find(lift.initial_floor);
+        auto initial_floor = std::string("");
+        auto target_elevation = std::numeric_limits<double>::min();
```

Comment:
The logic in the corner case here survived a long time without ever being caught since it's in the case where the initial floor is not specified, which doesn't happen in any of our pipelines.
This was supposed to be `max`, and below we would either iterate through all the floors and find the one with the minimum elevation, _or_ get the first element in the list. However, this snippet did both, and in the wrong way (`lift.floors` is an `unordered_map` with `std::string` as a key so I think `lift.floors.at(0)` would actually look for the lift named `0` and fail).
I addressed all of these in https://github.com/open-rmf/rmf_simulation/pull/114/commits/54b559ce4d8946f4ba488f1a229f35cf7383e383, where now the variable is initialized to `max` just to make sure the for loop will reliably find the minimum, then initialize the value at `0` if no floor is found as you pointed out.
I'm not 100% convinced with initializing `target_elevation` with 0 since doing a `if (floor.elevation < target_elevation)` would only trigger if the floor elevation was below 0, which might not be the case if we are only simulating upper floors.